components/error_code/src/storage.rs

```diff
@@ -45,6 +45,7 @@ define_error_codes!(
 
     PRIMARY_MISMATCH => ("PrimaryMismatch", "", ""),
     UNDETERMINED => ("Undetermined", "", ""),
+    GENERATION_OUT_OF_ORDER => ("GenerationOutOfOrder", "", ""),
```

Comment:
It makes sense not to return the error to client.

Reply:
There's a benefit returning the error to the client: if the client sends out-of-order value due to bugs, it will receive the error instead of continuing its incorrect behavior silently.

Reply:
> There's a benefit returning the error to the client: if the client sends out-of-order value due to bugs, it will receive the error instead of continuing its incorrect behavior silently.

But the only thing we can do in client side is still ignoring the error?

Reply:
This error should be dropped in the kv client in the common path:

1. requests with generation n sent
2. requests with generation n retried sending
3. requests with generation n done
4. requests with generation n+1 sent
5. `GenerationOutOfOrder` order responsed, ignored by client, because the requests are already canceled.

If the geneartion in client jumped back due to bugs, it may receive this error.

1. requests with generation n sent
2. requests with generation n retried sending
3. requests with generation n done
4. requests with generation n-1 sent
5. client got `GenerationOutOfOrder`